### PR TITLE
feat: add extra field to group

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

this pr adds a extra field when creating a group. user can define the content of any format but type of string for it.

### Rationale

This enriches the GROUP Usage scenario

### Example

NA

### Changes

Notable changes: 
* NA

### CODE REVIEW

File: x/storage/client/cli/tx.go [hunk 1]
```diff
@@ -516,12 +516,13 @@ func CmdDiscontinueObject() *cobra.Command {
 
 func CmdCreateGroup() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "create-group [group-name] [member-list]",
+		Use:   "create-group [group-name] [member-list [extra]",
```
Comment (guagualvcha) [inline x/storage/client/cli/tx.go:519]: What if a want to create a group with extra but without any member?

File: proto/greenfield/storage/types.proto [hunk 2]
```diff
@@ -104,6 +104,8 @@ message GroupInfo {
     (gogoproto.customtype) = "Uint",
     (gogoproto.nullable) = false
   ];
+  // extra is used to store extra info for the group
```
Comment (guagualvcha) [inline proto/greenfield/storage/types.proto:107]: We need to support update the extra field of a group.

File: e2e/tests/permission_test.go [hunk 3]
```diff
@@ -1233,14 +1233,14 @@ func (s *StorageTestSuite) TestGroupMembersAndPolicyGC() {
...
-	// policy is GC
-	_, err = s.Client.QueryPolicyById(ctx, &storagetypes.QueryPolicyByIdRequest{PolicyId: policyID.String()})
-	s.Require().Error(err)
-	s.Require().ErrorContains(err, "No such Policy")
+	//// the owner deletes the group
+	//msgDeleteGroup := storagetypes.NewMsgDeleteGroup(owner.GetAddr(), testGroupName)
+	//s.SendTxBlock(owner, msgDeleteGroup)
+	//
+	//// policy is GC
+	//_, err = s.Client.QueryPolicyById(ctx, &storagetypes.QueryPolicyByIdRequest{PolicyId: policyID.String()})
```
Comment (Fynnss) [inline e2e/tests/permission_test.go:1241]: Why comments these code?

File: x/storage/client/cli/tx.go (hunk 1 above)
Comment (Fynnss) [inline x/storage/client/cli/tx.go:519]: The extra is optional, so I suggest you use a flag for it.

File: proto/greenfield/storage/types.proto (hunk 2 above)
Comment (alexgao001) [reply proto/greenfield/storage/types.proto:107]: added

File: x/storage/types/message.go [hunk 4]
```diff
@@ -855,6 +855,7 @@ func NewMsgCreateGroup(creator sdk.AccAddress, groupName string, membersAcc []sd
 		Creator:   creator.String(),
 		GroupName: groupName,
 		Members:   members,
+		Extra:     extra,
```
[the remark this answers is not in the document]
Comment (alexgao001) [reply x/storage/types/message.go:859]: added limit

File: x/storage/client/cli/tx.go (hunk 1 above)
Comment (alexgao001) [reply x/storage/client/cli/tx.go]: fixed, 'extra' and 'member' can all be optional

File: e2e/tests/permission_test.go (hunk 3 above)
Comment (alexgao001) [reply e2e/tests/permission_test.go]: was for testing, removed,

File: x/storage/keeper/keeper.go [hunk 5]
```diff
@@ -1236,13 +1237,21 @@ func (k Keeper) UpdateGroupMember(ctx sdk.Context, operator sdk.AccAddress, grou
 		}
 
 	}
-	if err := ctx.EventManager().EmitTypedEvents(&types.EventUpdateGroupMember{
+
+	if groupInfo.Extra != opts.Extra {
```
Comment (guagualvcha) [inline x/storage/keeper/keeper.go:1241]: Could an empty`opts.Extra` means no update to `Extra`?

File: x/storage/types/message.go [hunk 6]
```diff
@@ -42,8 +42,11 @@ const (
 	TypeMsgDeleteGroup       = "delete_group"
 	TypeMsgLeaveGroup        = "leave_group"
 	TypeMsgUpdateGroupMember = "update_group_member"
+	TypeMsgUpdateGroupExtra  = "update_group_extra"
 	TypeMsgMirrorGroup       = "mirror_group"
 
+	MaxGroupExtraInfoLimit = 256
```
Comment (guagualvcha) [inline x/storage/types/message.go:48]: seems a bit short, please try 512.

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**